app/controllers/room/images_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Room
+  class ImagesController < ApplicationController
+    def create
+      new_media = Media.create
+      new_media.upload.attach(params[:hero_image_upload])
+      if room.update(hero_image: new_media)
+        flash[:notice] = t(".success", room_name: room.name)
+        redirect_to [:edit, room.space, room]
+      else
+        flash[:notice] = t(".failure", room_name: room.name)
+        render :new, status: :unprocessable_entity
+      end
+    end
+
+    helper_method def room
+      @room ||= if params[:room_id]
+        current_space.rooms.friendly.find(params[:room_id])
```

Comment:
I think you want to `policy_scope(Room).friendly.find(params[:room_id]`

I also think we don't need the `if` statement at all, since if there is no `:room_id` this will never be routed to, since it's nested within the `/rooms/:room_id/` path.